File: scala8/src/main/scala/fr/gblaquiere/scala8/function/HelloWorld.scala
package fr.gblaquiere.scala8.function

import javax.servlet.http.{HttpServlet, HttpServletRequest, HttpServletResponse}

class HelloWorld extends HttpServlet{
  // Only for Cloud Run compatibility
  override def doGet(req: HttpServletRequest, resp: HttpServletResponse): Unit = {
    return helloWorld(req, resp)
  }

  def helloWorld(req: HttpServletRequest, resp: HttpServletResponse): Unit = {
    resp.getWriter().write("Hello world!")
  }
}

File: kotlin8/src/main/kotlin/dev/gblaquiere/cloudruncompare/kotlin8/function/HelloWorld.kt
package dev.gblaquiere.cloudruncompare.kotlin8.function

import javax.servlet.http.*

class HelloWorld : HttpServlet() {

    // Only for Cloud Run compatibility
    override fun doGet(req: HttpServletRequest, resp: HttpServletResponse) {
        helloWorld(req,resp)
    }

    fun helloWorld(req: HttpServletRequest, resp: HttpServletResponse) {
        with(resp.writer) {
            println("Hello World!")
        }
    }
}
